casepro/msgs/forms.py

```diff
@@ -100,3 +100,34 @@ def clean_code(self):
     class Meta:
         model = Language
         fields = ('code', 'name', 'location')
+
+
+class FaqForm(forms.ModelForm):
+
+    question = forms.CharField(label=_("Question"), max_length=140)
+    answer = forms.CharField(label=_("Answer"), max_length=140)
+    language = forms.ModelChoiceField(queryset=Language.objects.filter())
+    # limit the parent choices to FAQs that have a ForeignKey parent that is None
+    parent = forms.ModelChoiceField(queryset=FAQ.objects.filter(parent=None), required=False)
+    labels = forms.ModelMultipleChoiceField(queryset=Label.objects.filter())
+
+    def clean_labels(self):
+        labels = self.cleaned_data['labels']
+
+        if 'parent' in self.cleaned_data:
+            parent = self.cleaned_data['parent']
+        else:
+            parent = None
+
+        if parent is not None:
+            parent_labels = parent.labels.all()
+            if list(parent_labels) != list(labels):
+                reqd_labels = ", ".join([str(label.name) for label in list(parent_labels)])
+                raise forms.ValidationError(
+                    "%s - %s" % (_("Labels should be the same as the parent FAQ's labels"), reqd_labels))
```

Comment:
I wonder if it's more logical to apply the labels for them if the parent is set?